### PR TITLE
Yes

### PR DESCRIPTION
**Pull request by joker-119**

### CODE REVIEW

File: Exiled.CustomRoles/API/Features/CustomRole.cs [hunk 1]
```diff
@@ -528,18 +534,18 @@ public virtual void AddRole(Player player)
...
-            if (position != Vector3.zero)
-            {
-                player.Position = position;
-            }
+                    Vector3 position = GetSpawnPosition();
+                    if (position != Vector3.zero)
+                    {
+                        player.Position = position;
+                    }
+                });
```
Comment (louis1706) [inline Exiled.CustomRoles/API/Features/CustomRole.cs:548]: you don't need to delay all this stuff

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Comment by NaoUnderscore** [comment]
I'm sure we're going to rework it as well, but as long as we keep it as is, this should be fine.
Just take a look to what @louis1706 is barking about.

File: Exiled.CustomRoles/API/Features/KeypressActivator.cs [hunk 2]
```diff
@@ -121,6 +99,34 @@ private IEnumerator<float> ProcessAltKey(Player player)
...
+
+            if (!coroutineTracker[ev.Player].IsRunning)
+                coroutineTracker[ev.Player] = Timing.RunCoroutine(ProcessAltKey(ev.Player));
+        }
+
+        private void OnEndingRound(EndingRoundEventArgs ev)
+        {
+            altTracker.Clear();
+            foreach (CoroutineHandle handle in coroutineTracker.Values)
+                Timing.KillCoroutines(handle);
```
Comment (NaoUnderscore) [inline Exiled.CustomRoles/API/Features/KeypressActivator.cs:126]: `Timing.KillCoroutines(coroutineTracker.Values.ToArray());`
May be better.

File: Exiled.CustomRoles/API/Features/CustomRole.cs (hunk 1 above)
Comment (joker-119) [reply Exiled.CustomRoles/API/Features/CustomRole.cs:548]: Ok, I'll close my or, let you remove them, and say I told you so when everything breaks. Have fun.

File: Exiled.CustomRoles/API/Features/KeypressActivator.cs (hunk 2 above)
Comment (louis1706) [reply Exiled.CustomRoles/API/Features/KeypressActivator.cs:126]: you just make me learn this BRUH

**Pull request closed without merging**